### PR TITLE
test(client): Fake environment cleanup

### PR DESCRIPTION
**Pull request by teogeb**
Call `FakeEnvironment#destroy` in each test which uses the fake environment.

The `destroy` method also calls `stop` method for each node. This doesn't change the functionality as the current implementations of `FakeNetworkNode` don't do any cleanup in `stop()`.

### CODE REVIEW

File: packages/client/test/integration/GroupKeyPersistence.test.ts [hunk 1]
```diff
@@ -24,11 +25,15 @@ describe('Group Key Persistence', () => {
     let storageNode: FakeStorageNode
     let environment: FakeEnvironment
 
-    beforeEach(() => {
+    beforeEach(async () => {
         environment = new FakeEnvironment()
         storageNode = environment.startStorageNode()
     })
```
Comment (harbu) [inline packages/client/test/integration/GroupKeyPersistence.test.ts]:
```suggestion
    beforeEach(() => {
        environment = new FakeEnvironment()
        storageNode = environment.startStorageNode()
    })
```

File: packages/client/test/integration/Stream.test.ts [hunk 2]
```diff
@@ -9,17 +9,19 @@ import { createTestStream } from '../test-utils/utils'
...
     let storageNode: FakeStorageNode
+    let environment: FakeEnvironment
 
-    beforeEach(() => {
-        const environment = new FakeEnvironment()
+    beforeEach(async () => {
+        environment = new FakeEnvironment()
         client = environment.createClient()
         storageNode = environment.startStorageNode()
     })
```
Comment (harbu) [inline packages/client/test/integration/Stream.test.ts]:
```suggestion
    beforeEach(() => {
        environment = new FakeEnvironment()
        client = environment.createClient()
        storageNode = environment.startStorageNode()
    })
```

File: packages/client/test/integration/events.test.ts [hunk 3]
```diff
@@ -6,16 +6,26 @@ describe('events', () => {
 
     describe('remove listeners when client destroyed', () => {
 
+        let environment: FakeEnvironment
+
+        beforeEach(async () => {
+            environment = new FakeEnvironment()
+        })
```
Comment (harbu) [inline packages/client/test/integration/events.test.ts]:
```suggestion
        beforeEach(() => {
            environment = new FakeEnvironment()
        })
```

File: packages/client/test/integration/pre-agreed-encryption-key.test.ts [hunk 4]
```diff
@@ -9,8 +9,17 @@ import { createTestStream } from '../test-utils/utils'
 
 describe('pre-agreed encryption key', () => {
 
+    let environment: FakeEnvironment
+
+    beforeAll(async () => {
+        environment = new FakeEnvironment()
+    })
```
Comment (harbu) [inline packages/client/test/integration/pre-agreed-encryption-key.test.ts]:
```suggestion
    beforeAll(() => {
        environment = new FakeEnvironment()
    })
```

**Comment by harbu** [comment]
Does `environment.destroy()` also destroy the `StreamrClient` instances? I noticed somewhere a `client.destroy()` was removed.

**Comment by teogeb** [comment]
> Does `environment.destroy()` also destroy the `StreamrClient` instances? I noticed somewhere a `client.destroy()` was removed.

Yes, `FakeEnvironment.ts#99.` Do you remember where it was destroyed?

**Review by harbu — APPROVED** [review]

**Pull request merged**